js/CobrowseIO.js

```diff
@@ -4,6 +4,84 @@ const NativeEventEmitter = require('react-native').NativeEventEmitter
 
 const emitter = new NativeEventEmitter(CobrowseIONative)
 
+class Session {
+  constructor (session) {
+    this._session = session || {}
+  }
+
+  get id () {
+    return this._session.id
+  }
+
+  get code () {
+    return this._session.code
+  }
+
+  get state () {
+    return this._session.state
+  }
+
+  get full_device () {
+    return this._session.full_device
+  }
+
+  get remote_control () {
+    return this._session.remote_control
+  }
+
+  get agent () {
+    return this._session.agent
+  }
+
+  activate () {
+    return CobrowseIONative.activateSession().then((session) => {
+      this._session = session
+    })
+  }
+
+  end () {
+    return CobrowseIONative.endSession()
+  }
+
+  hasAgent () {
+    return CobrowseIONative.sessionHasAgent()
+  }
+
+  isActive () {
+    return CobrowseIONative.sessionIsActive()
```

Comment:
For simple boolean methods like these, we don't need to proxy to the native module, we can just reimplement by checking the local session object. Otherwise we'll have ti implement all this on the binding in both Android and iOS

Reply:
I was torn, I initially started stubbing out checking the session object, but then I thought those properties would become out of date.

@headlessme How about adding a listener for session updates that updates the internal state of the session class? I'll implement checking the local session object regardless of your answer.

Reply:
Yea, I agree I think the session properties should update via the listener